### 1.
멘델 이론에 의하면 노란색 완두와 녹색 완두를 교배시키면 노란색과 녹색의 비가 3:1 이 된다고 한다. 
아래 표를 통해서 교배실험 결과가 멘델 법칙을 잘 따르고 있는지 유의수준 5% 하에서 검정하시오 



|노란색 완두 |녹색 완두|
|:---:|:---:|
|306|104|

In [3]:
import pandas as pd
import numpy as np
import warnings
import seaborn as sns
warnings.filterwarnings(action='ignore')

In [4]:
obs = pd.DataFrame({'본':[306,104],'이론':[300,100]})
obs.index = ['yellow','green']
obs.head()

본   이론
yellow  306  300
green   104  100

자료 분석:
교배실험의 결과인 완두들을 두가지 색상의 완두로 나누는 범주형자료이다 멘델 이론에 맞는지 적합도 검정이 필요하다. /독립성 검정을 이용한다.

귀무가설 H0 :교배실험 결과 노란 색상, 초록 색상의 완두 비가 멘델의 이론에 따라 3:1의 비를 따른다 . 

대립가설 H1:노란 색상 초록 색상의 완두비가 완두 비가 3:1 을 따르지 않는다. 

In [5]:
# 멘델의 이론에 따라 노란색과 초록색 완두의 비가 3:1 이므로 교배후 나올 노란색 완두의 비는 3/4 이다.
# one sample. chisquared test
from scipy.stats import chi2_contingency

chiresult = chi2_contingency(obs,correction = False)
print('Chi square: {}'.format(chiresult[0]))
print('p-value: {}'.format(chiresult[1]))

Chi square: 0.014382715172521963
p-value: 0.9045402168203751


### 2.

어느 교양 수업에 들어오는 학생들을 대상으로 운동(축구)을 잘하는 것과 학업성적이 
관련이 있는지에 대한 검정을 하고자 한다. 여기서, 성적은 총학점을 상, 중, 하로 나누었다.
유의수준 5% 하에서 검정하시오

|성적\운동|상|중|하|계|
|:---:|:---:|:---:|:---:|:---:|
|상|3|2|5|10|
|중|26|10|4|40|
|하|5|20|5|30|
|계|34|32|14|80|

In [66]:
#cross tab 없이
gh,gm,gl = [3,2,5] , [26,10,4] ,[5,20,5]
student =pd.DataFrame([gh,gm,gl], columns = ['ah','am','al'] , index =['gh','gm','gl'] )
student

ah  am  al
gh   3   2   5
gm  26  10   4
gl   5  20   5

자료 분석: 3개의 범주를 가진 두가지의 특성(학업성적, 운동 성적)이 서로 영향이 있는지 독립성 검정을 한다.

귀무가설 H0 :운동을 잘하는것과 학업성적은 관련성이 없다(독립이다)

대립가설 H1:운동을 잘하는 것과 학업성적릐 관련성이 있다(독립이 아니다)

In [65]:
chi2, p, dof, expected = chi2_contingency([gh,gm,gl])

msg = 'Test Statistic: {}\np-value: {}\nDegree of Freedom: {}'
print(msg.format(chi2, p, dof))
print(expected)

Test Statistic: 25.759803921568626
p-value: 3.537659661329406e-05
Degree of Freedom: 4
[[ 4.25  4.    1.75]
 [17.   16.    7.  ]
 [12.75 12.    5.25]]


In [51]:
data = pd.crosstab(student.columns,student.index)
data

col_0  gh  gl  gm
row_0            
ah      1   0   0
al      0   1   0
am      0   0   1

In [53]:
dtaset =  {'grade':['gh','gh','gh','gh','gh','gh','gh','gh','gh','gh'] ,
           'athle':['ah','ah','ah','am','am','al','al','al','al','al'] }

student_data = pd.DataFrame(dtaset,columns=['grade','athle'])
student_data

grade athle
0    gh    ah
1    gh    ah
2    gh    ah
3    gh    am
4    gh    am
5    gh    al
6    gh    al
7    gh    al
8    gh    al
9    gh    al

In [61]:
for x in range(10,36):
    student_data.loc[x] =['gm','ah']
for x in range(36,46):
    student_data.loc[x] =['gm','am']
for x in range(46,50):
    student_data.loc[x] =['gm','al']
    
for x in range(50,55):
    student_data.loc[x] =['gl','ah']
for x in range(55,75):
    student_data.loc[x] =['gl','am']
for x in range(75,80):
    student_data.loc[x] =['gl','al']    

In [70]:
result = pd.crosstab(student_data.grade,student_data.athle)
result

athle  ah  al  am
grade            
gh      3   5   2
gl      5   5  20
gm     26   4  10

In [71]:
# two sample chi-squared test
chi2_contingency(observed=result)

(25.759803921568626,
 3.537659661329406e-05,
 4,
 array([[ 4.25,  1.75,  4.  ],
        [12.75,  5.25, 12.  ],
        [17.  ,  7.  , 16.  ]]))

두가지 형식의로 카이제곱 검정을 돌려봄.

유의 확률 5 퍼인 0.05 보다 주어진 p-value 가 작으므로 유의하지 않다.
유의 확률(p-value) : 3.538e-05 = 3.538^-5=0.002

검정통계량(X-squared): 25.76 (자유도 df =4)

귀무가설 H0 :운동을 잘하는것과 학업성적은 관련성이 없다(독립이다)을기각한다.

운동을 잘하는것과 학업성적은 관련성이 있으므로 독립이 아니다 라고 할 수 있다.

### 3.
어느학과 3학년 학생들의 과대표에 대한만족도를 조사해보았다. 남녀 간의 만족도에 차이가 있는지를 유의수준 1% 하에서 검정하시오


|성별 / 만족도|만족|보통|불만족|
|:---:|:---:|:---:|:---:|
|남|15|4|1|
|여|10|8|3|



자료 분석: 속성 남,여를 가진 부모집단(subpopulation) 각각으로부터 만족도의 자료를 추출한 경우이며 분할표에서 부모집단의 비율이 동일한가를 검정한다.(동일성 검정)

귀무가설 H0 :남성과 여성의 만족도는 차이가 없다(동일하다) (p1j = p2j, j = 만족, 보통, 불만족)

대립가설 H1:H0가 아니다


In [73]:
boy,girl = [15,4,1] , [10,8,3]
student =pd.DataFrame([boy,girl], columns = ['good','normal','notgood'] ,
                      index =['boy','girl'] )
student

good  normal  notgood
boy     15       4        1
girl    10       8        3

In [74]:
chi2, p, dof, expected = chi2_contingency([boy,girl])

msg = 'Test Statistic: {}\np-value: {}\nDegree of Freedom: {}'
print(msg.format(chi2, p, dof))
print(expected)

Test Statistic: 3.310912698412698
p-value: 0.19100487089237705
Degree of Freedom: 2
[[12.19512195  5.85365854  1.95121951]
 [12.80487805  6.14634146  2.04878049]]



통계량 확인:

검정통계량(X-squared): 3.311 (자유도 df =2)

유의 확률(p-value) : 0.191

결론: 유의 확률인 0.191는 유의수준 1% = 0.01 보다 크므로 유의하다.

그렇므로 귀무 가설(H0)을 기각하지 않는다.

따라서, 남성과 여성의 만족도는 차이가 없다(동일하다) 라고 할 수있다.
